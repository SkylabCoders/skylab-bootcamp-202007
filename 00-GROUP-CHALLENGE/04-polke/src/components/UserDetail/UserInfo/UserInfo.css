.user-info-card {
	min-width: 75%;
	max-width: 75%;
	padding: 50px 10px 10px 10px;
	background-color: var(--bg-color-user-info-card);
	box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
	text-decoration: none !important;
	color: var(--font-color-user-info-card);
}
.user-info-card__main-title {
	align-self: center;
	font-weight: bolder;
}
.user-info-card img {
	align-self: center;
	width: 80%;
	border-radius: 50%;
}
.user-info-card:hover {
	box-shadow: 0 7px 10px rgba(0, 0, 0, 0.25), 0 5px 5px rgba(0, 0, 0, 0.22);
}

.user-info-card__name-title {
	text-align: center;
	padding-bottom: 3px;
	font-weight: bold;
	font-size: 1em;
	width: 100%;
}

.user-info-card__category {
	font-weight: bolder;
	padding-right: 5px;
}
