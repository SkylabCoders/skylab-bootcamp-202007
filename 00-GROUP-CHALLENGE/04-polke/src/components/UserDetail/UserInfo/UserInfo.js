import React, { useEffect, useState } from 'react';
import '../../../shared/generalStyles.css';
import './UserInfo.css';
import { Card } from 'react-bootstrap';
import userDetailStore from '../../../stores/userDetailStore';
import { loadUserImg } from '../../../actions/userDetailActions';

function UserInfo({ githubUserName }) {
	const [userImg, setuserImg] = useState([]);

	useEffect(() => {
		userDetailStore.addChangeListener(onChange);
		if (userImg.length === 0 && githubUserName) {
			loadUserImg(githubUserName);
		}

		return () => userDetailStore.removeChangeListener(onChange);
	}, [userImg.length]);

	function onChange() {
		setuserImg(userDetailStore.getUserInfo());
	}
	return (
		<Card className="user-info-card">
			<Card.Title className="user-info-card__main-title">USER INFO</Card.Title>
			<Card.Img roundedCircle variant="top" src={userImg.avatar_url} />
			<Card.Body>
				<Card.Title className="user-info-card__name-title">
					<a href={userImg.html_url}>{userImg.company}</a>
				</Card.Title>
				<Card.Text>
					<span className="user-info-card__category">Followers: </span>{' '}
					{userImg.followers}
				</Card.Text>
				<Card.Text>
					<span className="user-info-card__category">Bio: </span>
					{userImg.bio || 'No bio available'}
				</Card.Text>
				<Card.Text>
					<span className="user-info-card__category">Location: </span>{' '}
					{userImg.location || 'No location available'}
				</Card.Text>
			</Card.Body>
		</Card>
	);
}

export default UserInfo;
