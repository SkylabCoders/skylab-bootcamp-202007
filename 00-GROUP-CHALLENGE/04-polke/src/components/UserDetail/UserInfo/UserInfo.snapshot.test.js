import renderer from 'react-test-renderer';
import UserInfo from './UserInfo';
import React from 'react';

describe('UserInfo snapshot', () => {
	const tree = renderer.create(<UserInfo />);
	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
