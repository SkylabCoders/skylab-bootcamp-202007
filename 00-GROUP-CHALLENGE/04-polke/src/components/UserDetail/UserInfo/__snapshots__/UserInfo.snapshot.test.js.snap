// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo snapshot should match 1`] = `
<div
  className="user-info-card card"
>
  <div
    className="user-info-card__main-title card-title h5"
  >
    USER INFO
  </div>
  <img
    className="card-img-top"
    roundedCircle={true}
  />
  <div
    className="card-body"
  >
    <div
      className="user-info-card__name-title card-title h5"
    >
      <a />
    </div>
    <p
      className="card-text"
    >
      <span
        className="user-info-card__category"
      >
        Followers: 
      </span>
       
    </p>
    <p
      className="card-text"
    >
      <span
        className="user-info-card__category"
      >
        Bio: 
      </span>
      No bio available
    </p>
    <p
      className="card-text"
    >
      <span
        className="user-info-card__category"
      >
        Location: 
      </span>
       
      No location available
    </p>
  </div>
</div>
`;
